packages/xstate-vue/src/xstate-machine.ts

```diff
@@ -0,0 +1,103 @@
+import { interpret, State } from 'xstate';
+import Vue, { VNode } from 'vue';
+
+export default Vue.extend({
+  render(h): VNode {
+    const { current, service, send } = this;
+    return h('div', [
+      this.$scopedSlots.default({
+        current,
+        service,
+        send
+      })
+    ]);
+  },
+  props: {
```

Comment:
is this needed? seems like a runtime type checking thing? should we just rely on static type checker with this sort of thing?

Reply:
this is the default way 
https://vuejs.org/v2/guide/components-props.html#Prop-Validation

I think with typescript it would be `Object as () => SomeType`